### ISSUE TITLE
Multi-repo and Image builder changes

### ISSUE THREAD

**Issue by daxmc99** [rancher/rio#498]
Rio needs to be able to support multi-repo deployments. 

Example flow for this issue:
1. Multiple images builds are specified in service 
2. Rio will build all the images
3. Rio waits for all images to be built
4. Rio creates a k8s deployment

Ex 2:
1. ImageBuilder CRD is created
2. ImageBuilder controller will see new CRD, create a TaskRun for Tekton using Buildkit
3. ImageBuilder controller will check if the build is done (OnChange?)
4. Once build finishes, the controller will update Rio service. The `build` field being set signals to the service that the image is built.

This flow already exists, but only with a single image. 

Nice to have; but possibly outside scope of this issue:
1. Also support mixed configuration of ImageBuilds and pre-built images (ie: `image: nginx`)

Merge to api-refactor branch

**Comment by StrongMonkey** [comment]
Also need to implement a controller that watch ImageBuild resource and only build images.

**Comment by rancher-max** [comment]
When doing the first example flow:
It worked with just adding a docker image to the service spec, yielding a k8s deployment with both the initial GH container and the Docker container specified.

When trying to add another GH container to the spec, it built successfully (output from both `kubectl get taskruns` and `rio build history`) but `kubectl get deployment` did not have the resource anymore, and `rio ps` had a detail of `keen-nobel5: not ready; keen-nobel5 waiting on build`

**Comment by StrongMonkey** [comment]
@rancher-max Can you re-test this on the latest alpha?

**Comment by rancher-max** [comment]
ImageBuilder is different in v0.6.0-alpha.2. Verifying this is checking that `rio build history` shows the run from `kubectl get taskruns. Also, running `rio inspect` on an output from that and `kubectl get taskruns <name> -o yaml` produce the same result. This works as expected in v0.6.0-alpha.2.

Still need to verify multi-repo

**Comment by rancher-max** [comment]
MultiRepo still failing to build separate github containers, but builds docker images on v0.6.0-alpha.2

Riofile:
```
services:
  test2:
   template: true
   build:
    repo: https://github.com/rancher-max/rio-demo
    branch: master
   containers:
   - name: sidecar1
     build:
      repo: https://github.com/rancher-max/rio-demo
      branch: k3s-branch
   - name: sidecar2
     image: ibuildthecloud/demo:v1
     scale: 2
     ports: 8080/http
   - name: sidecar3
     build:
      repo: https://github.com/rancher-max/rio-demo
      branch: max-test
```

`rio -n issue-ns up -f /path/to/Riofile`

`rio -n issue-ns ps` --> (two different outputs, depending on timing)
```
NAME                  IMAGE                                ENDPOINT                                                 SCALE     APP/VERSION      WEIGHT    CREATED              DETAIL
test2-ccd1159-d7239   issue-ns-test2-ccd1159-d7239:5e18e   https://test2-fbac4467-issue-ns.xxxxxx.on-rio.io:32269   0/1 0%    test2/fbac4467   100%      About a minute ago   sidecar1(CrashLoopBackOff/back-off 20s restarting failed container=sidecar1 pod=test2-ccd1159-d7239-b87b6bd79-px67m_issue-ns(59c8f410-5d0e-4ef6-8528-74e02c3804d5)); sidecar3(CrashLoopBackOff/back-off 20s restarting failed container=sidecar3 pod=test2-ccd1159-d7239-b87b6bd79-px67m_issue-ns(59c8f410-5d0e-4ef6-8528-74e02c3804d5))
```
AND
```
NAME                  IMAGE                                ENDPOINT                                                 SCALE     APP/VERSION      WEIGHT    CREATED              DETAIL
test2-ccd1159-d7239   issue-ns-test2-ccd1159-d7239:5e18e   https://test2-fbac4467-issue-ns.xxxxxx.on-rio.io:32269   0/1 0%    test2/fbac4467   100%      About a minute ago   sidecar1(Error/exit code not zero), exit code: 1;sidecar3(Error/exit code not zero), exit code: 1
```

`rio -n issue-ns build history` -->'
```
NAME                                               SERVICE/STACK   REVISION                                   CREATED         SUCCEED   REASON
taskrun/issue-ns/sidecar3-ccd11-09e4d                              076f9e67b817b4d569f2755dbab0d1eb7c36ca84   3 minutes ago   True      Succeeded
taskrun/issue-ns/test2-ccd1159-d7239-ccd11-d7239                   5e18e82022a1bc251941df338a3a74727fbfca86   3 minutes ago   True      Succeeded
taskrun/issue-ns/sidecar1-ccd11-b1187                              beaf76d94d3906e6adae50e7ae3fb6ea9df9bf73   3 minutes ago   True      Succeeded
```

Attempted on both GKE and RKE with k8s 1.16
`kubectl version` -->
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by rancher-max** [comment]
^^^^ That comment was because all of the branches in the repo defined were shipping on 8080, causing conflicts. After changing that, everything works as expected with v0.6.0-alpha.2
